src/components/common/DiscountCode.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+  <div class="row">
+    <div class="col-sm-offset-9 col-sm-3">
+      <ServerError :error="serverError">
+        <template slot-scope="{ graphQLError }">
+          {{ getErrorMessage(graphQLError) }}
+        </template>
+      </ServerError>
+    </div>
+    <div class="col-sm-12">
+      <form @submit.prevent="submit" class="text-right promotion-info">
+        <span class="text-uppercase promo-info-text">
+          {{ $t('promoInfo') }}
+          <img src="../../assets/img/information.png"
+               class="info-icon"
+               alt="information icon">
+        </span>
+        <input id="promo-code" type="text" name="code" v-model="code"
+               data-test="discount-code-input">
+        <LoadingButton :buttonState="buttonState"
+                       class="apply-button"
+                       data-test="apply-discount-code-button">
+          {{ $t('apply') }}
+        </LoadingButton>
+        <div v-if="cartLike.discountCodes.length > 0"
+             class="row"
+             style="margin-top: 15px">
+          <div class="col-sm-12">
+            <div class="text-right order-discount col-sm-offset-5">
+              <span class="col-sm-3"
+                    style="font-weight: bold">
+                Applied discounts:
+              </span>
+            </div>
+            <div v-for="discountInfo in cartLike.discountCodes"
+                :key='discountInfo.discountCode.id'
+                class="text-right order-discount col-sm-offset-7">
+              <div class="col-sm-3"
+                   data-test="discount-code-name">
+                {{ discountInfo.discountCode.code }}
+              </div>
+              <div class="col-sm-3">{{ discountInfo.discountCode.name }}</div>
+              <div class="col-sm-5">
+                ({{ discountInfo.discountCode.description }})
```

Comment:
Now I see in DiscountCode entity there's `name` and `description`. I thought there was only `code` and `description`.
So, where you used `name`, you should use `code` (e.g. `SUNRISE`, `CTXMAS19`), and then where you used `description` maybe we can use `name`, which should be shorter and more similar to what we are looking for than `description` (of course in the end it's always up to the merchant). 

Remember that what we are looking for is:
CTXMAS19 .....  50% off